public/app/plugins/datasource/prometheus/querybuilder/components/MetricEncyclopediaModal.tsx

```diff
@@ -260,14 +263,14 @@ export const MetricEncyclopediaModal = (props: Props) => {
         // backend and frontend
         // skipLetterSearch is purely for setting letter spans as active so they can be clicked
         // where we don't filter out by letter search and keep letters of all other filters
-        if (letterSearch && !skipLetterSearch) {
+        if ((keepMetric || !fuzzySearchQuery) && letterSearch && !skipLetterSearch) {
```

Comment:
@gtk-grafana I need to come up with a set of rules for the filtering. That will make it easier to keep track of the order of things. For now, it's 1) fuzzy search, 2) alphabet, 3) filter types. I also need to consider the metrics that don't have metadata, they are not filtered out by type unless the switch is on to filter them.